Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp

```diff
@@ -281,6 +281,52 @@ static void parse_alias_list(Core::File& file, PropList const& prop_list, Vector
     }
 }
 
+static void parse_name_aliases(Core::File& file, Vector<CodePointData>& code_point_data)
+{
+    auto iterator = code_point_data.begin();
+    VERIFY(!iterator.is_end());
+    // We use a single iterator because both lists should be sorted, and repeated lookups are unnecessary.
+
+    auto code_point_data_at = [&iterator](u32 code_point) -> CodePointData& {
+        VERIFY(!iterator.is_end());
+        while (iterator->code_point < code_point) {
+            ++iterator;
+            VERIFY(!iterator.is_end());
+        }
+        VERIFY(iterator->code_point == code_point);
+        return *iterator;
+    };
+
+    // We always use the *first* "control"-type alias. The actual reason for this is to avoid the silly name "END OF LINE" for code point 0x000a.
+    u32 last_overridden = (u32)-1;
+
+    while (file.can_read_line()) {
```

Comment:
This is a bit posthumous now, but I think this could have been quite simpler by parsing NameAliases.txt before UnicodeData.txt. 

You could add a `HashMap<u32, String> name_aliases` to the `UnicodeData` struct to avoid doing lookups via `code_point_data_at` entirely. And check `name_aliases.contains(code_point)` instead of tracking this via `last_overriden`.

Then in `parse_unicode_data`, you would do:
```c++
data.name = unicode_data.name_aliases.get(data.code_point).value_or(segments[1]);
```

(or something equivalent)

Reply:
Loading the corrections before the thing that is being corrected seems backwards to me.

Also, note that `code_point_data_at` is not really a lookup; the loop just traverses both lists at the same time. I don't think that adding more HashMaps and lookups really is "simpler".

Reply:
> Loading the corrections before the thing that is being corrected seems backwards to me.

Doesn't seem backwards to me at all.

> I don't think that adding more HashMaps and lookups really is "simpler".

I don't see how a single hash map isn't simpler than maintaining an iterator, a traversal lambda, and the last code point.

```c++
static void parse_name_aliases(Core::File& file, UnicodeData& unicode_data)
{
    while (file.can_read_line()) {
        auto line = file.read_line();
        if (line.is_empty() || line.starts_with('#'))
            continue;

        auto segments = line.split_view(';', true);
        VERIFY(segments.size() == 3);

        auto code_point = AK::StringUtils::convert_to_uint_from_hex<u32>(segments[0].trim_whitespace());
        auto alias = segments[1].trim_whitespace();
        auto reason = segments[2].trim_whitespace();

        VERIFY(!alias.is_empty());
        if (!reason.is_one_of("correction"sv, "control"sv))
            continue;

        if (!unicode_data.contains(*code_point))
            unicode_data.name_aliases.set(*code_point, alias);
    }
}
```

It literally halves the parser.